vaadin-combo-box-flow/src/main/resources/META-INF/resources/frontend/comboBoxConnector.js

```diff
@@ -25,12 +25,35 @@ window.Vaadin.Flow.comboBoxConnector = {
 
     comboBox.$connector = {};
 
+    // holds pageIndex -> callback pairs of subsequent indexes (current active range)
     let pageCallbacks = {};
     let cache = {};
     let lastFilter = '';
+    const placeHolder = new Vaadin.ComboBoxPlaceholder();
+    const MAX_RANGE_COUNT = Math.max(comboBox.pageSize * 2, 500); // Max item count in active range
+
+    let setRequestedRange;
+
+    const clearPageCallbacks = (pages = Object.keys(pageCallbacks)) => {
```

Comment:
Done. Added a call to `refreshValue()` to `setRequestedRange` and a test case to cover this case.

Reply:
Looks like `refreshValue()` will not only update the component's value but also override any filters a user might have typed to the field. Will need to find a way to avoid this to allow filtering while a value is selected.

Reply:
Ok changed the logic to maintain the selectedItem key (and value) in the connector instead. Added a test to cover the mentioned regression.